### PR TITLE
Add e2e test for hana scale up cost optimized scenario

### PR DESCRIPTION
**Pull request by EMaksy**
# Description

This PR is an ad-don to the newly introduced[ hana scale up scenario discovery ](https://github.com/trento-project/web/pull/3152). 
Besides the front and back end tests this pr introduces e2e tests.

Added a new cluster called hana_cost_opt which contains a Database called HDC and QAS
image

This scenario adds a hana cost opt scenario
image

## What has been changed ? 

- Refactor of the HostsList to remove the line break, this helps with the e2e tests and aligns like in other parts of the code
- Updated existing tests in order to check for multiple SID's which a cost opt scenario has
- Updated index of old tests as new hosts were added in the scenario
- Added new test for hana scale up cost opt
- Added two new nodes scenarios 2372b24f-3d7a-5d01-9b1a-a2c4c95c53d4 and fa7a5602-232b-5389-96a6-f5f5de6ff9a2

### CODE REVIEW

File: assets/js/pages/HostsList/HostsList.jsx [hunk 1]
```diff
@@ -119,22 +119,18 @@ function HostsList() {
...
+            <span key={`${sids[index]}-${instance?.id}`}>
+              <SapSystemLink
+                key={`${getInstanceID(instance)}-${instance?.id}`}
+                systemType={instance?.type}
+                sapSystemId={getInstanceID(instance)}
+              >
+                {instance?.sid}
+              </SapSystemLink>{' '}
+            </span>
+          ));
```
Comment (dottorblaster) [inline assets/js/pages/HostsList/HostsList.jsx:133]: I'm not sure a PR introducing some tests should actually modify features' code too :smile: 


This refactor is just made to remove the line break right?

**Review by dottorblaster — COMMENTED** [review]
The PR looks good, I just wanted to think more about the fact that we're turning the whole `27-nodes` landscape into `29-nodes`. I don't know if it's something we can avoid due to how the tests are structured.

@balanza @CDimonaco do you have any idea about any other possible approach?

File: assets/js/pages/HostsList/HostsList.jsx (hunk 1 above)
Comment (EMaksy) [reply assets/js/pages/HostsList/HostsList.jsx:133]: Did the small refactor to remove the line break, it produced some issues with existing tests when additional sid was added. 
I guess i can revert it, but i may to modify existing tests more. 
What do you think ?

**Comment by EMaksy** [comment]
The main reason for renaming the healthy-27-node-SAP-cluster  scenario to healthy-29-node-SAP-cluster is that we added two new nodes.. tbh i never thought about the name of healthy-27-node-SAP-cluster before i saw that it contains 27 systems. 
An alternative approach would be to keep the 27 scenario and add a new 29 scenario. Use only the healthy-29-node-SAP-cluster scenario where required, but it is at a lot of points tbh so not sure why to keep the duplicate.

**Comment by EMaksy** [comment]
Actually it makes a lot of sense just to remove the number tbh of the scenario. 
@CDimonaco any additional ideas before i start making changes ?

File: assets/js/pages/HostsList/HostsList.jsx (hunk 1 above)
Comment (CDimonaco) [reply assets/js/pages/HostsList/HostsList.jsx:133]: This change have some ui changes? Like in the appaerence of things

File: test/e2e/cypress/e2e/clusters_overview.cy.js [hunk 2]
```diff
@@ -54,7 +54,9 @@ context('Clusters Overview', () => {
           .contains('th', 'SID')
           .invoke('index')
           .then((i) => {
-            cy.get('@clusterRow').eq(i).should('contain', cluster.sid);
+            cluster.sid.forEach((sid) => {
```
Comment (CDimonaco) [inline test/e2e/cypress/e2e/clusters_overview.cy.js:57]: Maybe `sids` instead of sid?

File: test/e2e/cypress/e2e/hana_cluster_details.cy.js [hunk 3]
```diff
@@ -233,6 +234,102 @@ context('HANA cluster details', () => {
...
+        .next()
+        .contains(availableHanaClusterCostOpt.provider);
+    });
+
+    it('should have all cost optimized SID`s and correct database links', () => {
+      cy.get('.tn-cluster-details')
+        .contains('SID')
+        .parent()
+        .within(() => {
+          availableHanaClusterCostOpt.sid.forEach((sid, index) => {
```
Comment (CDimonaco) [inline test/e2e/cypress/e2e/hana_cluster_details.cy.js:259]: sids as the previous one

File: test/e2e/cypress/e2e/hosts_overview.cy.js [hunk 4]
```diff
@@ -140,13 +140,19 @@ context('Hosts Overview', () => {
...
           .then((i) => {
             if (host.clusterId) {
-              cy.get('@hostRow').eq(i).should('contain', host.sapSystemSid);
-              cy.get('@hostRow').eq(i).click();
-              cy.location('pathname').should(
-                'eq',
-                `/databases/${host.sapSystemId}`
-              );
-              cy.go('back');
+              host.sapSystemSid.forEach((sid, counter) => {
```
Comment (CDimonaco) [inline test/e2e/cypress/e2e/hosts_overview.cy.js:143]: prural

File: test/e2e/cypress/fixtures/clusters-overview/available_clusters.js [hunk 5]
```diff
@@ -2,55 +2,61 @@ export const availableClusters = [
   {
     id: '8a66f8fb-5fe9-51b3-a34c-24321271a4e3',
     name: 'drbd_cluster',
-    sid: '',
+    sid: [''],
```
Comment (CDimonaco) [inline test/e2e/cypress/fixtures/clusters-overview/available_clusters.js:5]: ofc prurals also here

**Review by CDimonaco — COMMENTED** [review]
Just left some comments

**Comment by CDimonaco** [comment]
I think we can stick with the numbers of nodes in the scenario name, we can think about it later.
Just to keep the convention alive until we don't need it anymore.

**Review by dottorblaster — APPROVED** [review]
LGTM, just apply comments by @CDimonaco and that's :fire:

File: assets/js/pages/HostsList/HostsList.jsx (hunk 1 above)
Comment (EMaksy) [reply assets/js/pages/HostsList/HostsList.jsx:133]: No it looks the same and that is the fun thing about it :+1:

**Pull request merged**